assets/src/edit-story/components/canvas/selectionCanvas.js

```diff
@@ -0,0 +1,149 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useStory } from '../../app';
+import useCanvas from '../canvas/useCanvas';
+import withOverlay from '../overlay/withOverlay';
+import InOverlay from '../overlay';
+
+const LasoMode = {
+	OFF: 0,
+	ON: 1,
+	MAYBE: 2,
```

Comment:
What's the "maybe" for?

Reply:
The sequence we are looking for is pointer down -> move -> up. So the current algo is:

1. Ignore all "move" if there pointer is not down.
2. Once "down" arrives, we are still not sure if we are in the laso mode until we see some "move" events. Switch to MAYBE mode.
3. Once "move" arrives and we're in MAYBE mode, switch to ON mode.
4. Once "up" arrives, switch to OFF mode. If selection is valid, select the intersecting elements.

Further with possible drift we may, at some point, want to introduce a "move" threshold to control for false positives, e.g. how `Moveable` does. But for now it didn't seem necessary to me.

Reply:
Yep, makes sense, perhaps it would be good to leave a comment here that "maybe" is for when the mouse is down but no move events have happened yet.